src/views/Package/PackageState.tsx

```diff
@@ -104,6 +283,37 @@ export const PackageStateProvider: FunctionComponent = ({ children }) => {
       (assemblyResponse.loading || markdownResponse.loading)
   );
 
+  const [processedMd, setProcessMd] = useState<string | undefined>();
+
+  useEffect(() => {
+    let isMounted = true;
+    if (markdownResponse.data) {
+      void remark()
+        .use(emoji)
+        .process(markdownResponse.data)
+        .then((file) => {
+          if (isMounted) {
+            setProcessMd(String(file));
+          }
+        });
+    }
+
+    return () => {
+      isMounted = false;
+    };
+  }, [markdownResponse]);
```

Comment:
I did some profiling on some of the larger docs like `@aws-cdk/aws-ec2` that still take around 20-30 seconds to load, and it seems like this useEffect call is the main bottleneck. If I remove this (and instead pass `markdownResponse.data` to `parseMarkdownStructure` below), then the docs load fast as expected. Let's see if there's a way to defer processing emoji instead of doing it up front

Reply:
yeah adding emoji support was done last minute once I noticed it breaking on construct-hub. Wasn't expecting a remark parse with a single plugin to be this slow, but I suspect it is `remark-emoji` itself and stacking `Array.find`s for each emoji encountered. https://github.com/rhysd/remark-emoji/blob/master/index.js#L21 Will look into handling the replacement ourselves.

Reply:
Fixed by manually handling emoji rendering in menu items from header strings using `node-emoji`.